crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -36,26 +36,32 @@ def restore
 
       protected
 
-      def knife
-        [:nodes, :roles, :clients, :databags].each do |type|
-          Dir.glob(@data.join("knife", type.to_s, "**", "*")).each do |file|
-            file = Pathname.new(file)
-            next unless file.extname == ".json"
-            record = JSON.load(file.read)
-            if file.basename.to_s.include?("-default.json")
-              bc_name = file.basename.to_s.split("-").first
-              Proposal.create(barclamp: bc_name, name: "default", properties: json)
-              SchemaMigration.run_for_bc(bc_name)
-            else
-              record.save
+      def restore_chef
+        begin
+          [:nodes, :roles, :clients, :databags].each do |type|
+            Dir.glob(@data.join("knife", type.to_s, "**", "*")).each do |file|
+              file = Pathname.new(file)
+              next unless file.extname == ".json"
+              record = JSON.load(file.read)
+              if file.basename.to_s.include?("-default.json") && type == :databags
```

Comment:
There's a problem with this. Not all proposal end in ```-default.json```. All proposals that can have multiple instances have the instance name here, instead of ```default```. This is especially true for the pacemaker proposals (but there are a few others as well)